src/mlpack/core/math/random.hpp

```diff
@@ -88,30 +88,22 @@ inline int RandInt(const int lo, const int hiExclusive)
  */
 inline int RandInt(const arma::vec& weights)
 {
-  // Build cumulative probabilities from event probabilities.
-  std::vector<double> slots(weights.size());
-  double runningSum = 0;
-  for (size_t i = 0; i < weights.size(); ++i) {
-    double el = weights(i);
-    if (el < 0) {
-      std::ostringstream oss;
+  // Check that constraints on weights parameter are satisfied.
+  if (arma::min(weights) < 0) {
```

Comment:
I know I'm really picky about the style, can you go through the code and adjust the lines with the following pattern: ``) {``.